projects/dotcms-field-elements/src/components/dot-date/dot-date.e2e.ts

```diff
@@ -1,131 +1,327 @@
-        expect(required).toBeFalsy();
-    });
+            it('should be valid, untouched & pristine and required when filled on load', async () => {
+                dotTestUtil.triggerStatusChange(true, false, true, inputCalendar);
+                await page.waitForChanges();
+                expect(element).toHaveClasses(dotTestUtil.class.filledRequiredPristine);
+            });
 
-    it('should show invalid range validation message', async () => {
-        element.setProperty('value', '2015-10-01');
-        await input.press('2');
-        await page.waitForChanges();
-        const errorMessage = await page.find('.dot-field__error-message');
-        expect(errorMessage.innerHTML).toBe('Invalid Date Range');
-    });
+            it('should be valid, touched & dirty and required when filled', async () => {
+                dotTestUtil.triggerStatusChange(false, true, true, inputCalendar);
+                await page.waitForChanges();
+                expect(element).toHaveClasses(dotTestUtil.class.filledRequired);
+            });
 
-    it('should set the default value of "min" when is not valid', async () => {
-        element.setProperty('min', { test: 'min' });
-        await page.waitForChanges();
-        expect(await input.getProperty('min')).toBe('');
+            it('should be invalid, untouched, pristine and required when empty on load', async () => {
+                dotTestUtil.triggerStatusChange(true, false, false, inputCalendar);
+                await page.waitForChanges();
+                expect(element).toHaveClasses(dotTestUtil.class.emptyRequiredPristine);
+            });
+
+            it('should be invalid, touched, dirty and required when valued is cleared', async () => {
+                dotTestUtil.triggerStatusChange(false, true, false, inputCalendar);
+                await page.waitForChanges();
+                expect(element).toHaveClasses(dotTestUtil.class.emptyRequired);
+            });
+
+            it('should have touched but pristine on blur', async () => {
+                dotTestUtil.triggerStatusChange(true, true, true, inputCalendar);
+                await page.waitForChanges();
+                expect(element).toHaveClasses(dotTestUtil.class.touchedPristine);
+            });
+        });
     });
 
-    it('should set the default value of "max" when is not valid', async () => {
-        element.setProperty('max', [1, 6, 7]);
-        await page.waitForChanges();
-       expect(await input.getProperty('max')).toBe('');
+    describe('@Props', () => {
+        describe('value', () => {
+            it('should render default value', () => {
+                expect(inputCalendar.getAttribute('value')).toBe('');
+            });
+
+            it('should pass correctly to dot-input-calendar', async () => {
+                element.setProperty('value', '2019-01-01');
+                await page.waitForChanges();
+                expect(await inputCalendar.getProperty('value')).toBe('2019-01-01');
```

Comment:
getAttribute?

Reply:
actually I tried that, but can't explain why `getAttribute` do not work in this scenario.